File: swagger/models/get_staff_appointments_request.py
from .base_model import BaseModel


class GetStaffAppointmentsRequest(BaseModel):
	"""
	no description available

	Attributes:

		appointment_ids
		A list of the requested appointment IDs.

		location_ids
		A list of the requested location IDs.

		start_date
		The start date of the requested date range. If omitted, the default is
		used.   <br />Default: **today’s date**

		end_date
		The end date of the requested date range.   <br />Default:
		**StartDate**

		staff_ids
		List of staff IDs to be returned. Use a value of zero to return all
		staff appointments.

		client_id
		The client ID to be returned.

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'AppointmentIds': 'appointment_ids',
		'LocationIds': 'location_ids',
		'StartDate': 'start_date',
		'EndDate': 'end_date',
		'StaffIds': 'staff_ids',
		'ClientId': 'client_id',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'appointment_ids': 'AppointmentIds',
		'location_ids': 'LocationIds',
		'start_date': 'StartDate',
		'end_date': 'EndDate',
		'staff_ids': 'StaffIds',
		'client_id': 'ClientId',
		'limit': 'Limit',
		'offset': 'Offset',
		}

File: swagger/Models/GetStaffAppointmentsRequest.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class GetStaffAppointmentsRequest extends BaseModel
{
	/**
	 * A list of the requested appointment IDs.
	 * 
	 * @var integer[] 
	 */
	public $appointmentIds;
	/**
	 * A list of the requested location IDs.
	 * 
	 * @var integer[] 
	 */
	public $locationIds;
	/**
	 * The start date of the requested date range. If omitted, the default is
	 * used.   <br />Default: **today’s date**
	 * 
	 * @var string format:date-time
	 */
	public $startDate;
	/**
	 * The end date of the requested date range.   <br />Default:
	 * **StartDate**
	 * 
	 * @var string format:date-time
	 */
	public $endDate;
	/**
	 * List of staff IDs to be returned. Use a value of zero to return all
	 * staff appointments.
	 * 
	 * @var integer[] 
	 */
	public $staffIds;
	/**
	 * The client ID to be returned.
	 * 
	 * @var string 
	 */
	public $clientId;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer format:int32
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer format:int32
	 */
	public $offset;

	protected function getInputMap()
	{
		return [
			'AppointmentIds' => ['appointmentIds', 'array', null],
			'LocationIds' => ['locationIds', 'array', null],
			'StartDate' => ['startDate', 'string', 'date-time'],
			'EndDate' => ['endDate', 'string', 'date-time'],
			'StaffIds' => ['staffIds', 'array', null],
			'ClientId' => ['clientId', 'string', null],
			'Limit' => ['limit', 'integer', 'int32'],
			'Offset' => ['offset', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'appointmentIds' => ['AppointmentIds', 'array', null],
			'locationIds' => ['LocationIds', 'array', null],
			'startDate' => ['StartDate', 'string', 'date-time'],
			'endDate' => ['EndDate', 'string', 'date-time'],
			'staffIds' => ['StaffIds', 'array', null],
			'clientId' => ['ClientId', 'string', null],
			'limit' => ['Limit', 'integer', 'int32'],
			'offset' => ['Offset', 'integer', 'int32'],
			];
	}
}
